lib/core/scheduler.rb

```diff
@@ -78,6 +78,13 @@ def sync_triggered?(connector_settings)
         return false
       end
 
+      # We want to sync when sync never actually happened
+      last_synced = connector_settings[:last_synced]
+      if last_synced.nil? || last_synced.empty?
+        Utility::Logger.info("Connector #{connector_settings.id} has never synced yet, running initial sync.")
+        return true
+      end
```

Comment:
If it's never synced, we should sync it only if sync is enabled. therefore I move this block after the `scheduling_settings[:enabled]` check.

Reply:
Ha, good catch!